addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -17,22 +19,24 @@ options.registry.SocialMedia = options.Class.extend({
             return [generateHTMLId(), el];
         }));
         await this._super(...arguments);
-        let websiteId;
-        this.trigger_up('context_get', {
-            callback: function (ctx) {
-                websiteId = ctx['website_id'];
-            },
-        });
-        // Fetch URLs for db links.
-        [this.dbSocialValues] = await this._rpc({
-            model: 'website',
-            method: 'read',
-            args: [websiteId, ['social_facebook', 'social_twitter', 'social_linkedin',
+        if (!dbSocialValues) {
+            let websiteId;
+            this.trigger_up('context_get', {
+                callback: function (ctx) {
+                    websiteId = ctx['website_id'];
+                },
+            });
+            // Fetch URLs for db links.
+            [dbSocialValues] = await this._rpc({
```

Comment:
It's fine to cache this that way in this case, but it is not perfect so fix this for good practice:
In your implementation, if you somehow drop two snippets "at the same time", you'll still end up doing two RPC.

- `dbSocialValues` is falsy
- Drop first snippet
- Start rpc
- Drop second snippet
- `dbSocialValues` is still falsy -> start new rpc
- First rpc is finished -> `dbSocialValues` is set
=> Fail

You want to do the same thing you did but not use the `dbSocialValues` variable to determine if you must start a new RPC -> save the actual Promise of the rpc. So something like

```javascript
let dbSocialValues;
let dbSocialValuesProm;

// [...]

if (!dbSocialValuesProm) {
    dbSocialValuesProm = this._rpc(/*... */).then(values => dbSocialValues = values);
}
await dbSocialValuesProm;
```

=> The two snippets drop will await the same Promise, only one RPC will be made.